### PR TITLE
Fix stale dataflow prebuilt image cleaner

### PR DESCRIPTION
**Pull request by Abacn**
Two efficiency improvement

- For the nonzero sized images themself (creation time not in epoch), do not invoke `docker manifest inspect`
- For images in us.gcr.io (that are for testing purpose), decrease the retention time to 3 days so there are fewer zero-sized image that need to check digest and decide the actual creation time each time the workflow is run

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Abacn** [comment]
running: https://github.com/apache/beam/actions/runs/7730912155/job/21077372492

**Comment by Abacn** [comment]
On the other hand we are leaking too many test containers. Opened #30173

**Comment by Abacn** [comment]
R: @damccorm this is following the same solution of #30153 that allows the script run on completion

File: .test-infra/tools/stale_dataflow_prebuilt_image_cleaner.sh [hunk 1]
```diff
@@ -89,24 +98,24 @@ for image_name in ${IMAGE_NAMES[@]}; do
...
+            FAILED_IMAGES+=" $current"
             continue
           fi
           SHOULD_DELETE=0
           DIGEST=$(echo $MANIFEST |  jq -r '.manifests[0].digest')
           if [ "$DIGEST" != "null" ]; then
             SHOULD_DELETE=1
-            for i in ${STALE_IMAGES_CURRENT[@]}; do
-              if [ "$i" = "$DIGEST" ]; then
+            for (( j_stale_images_current=0; j_stale_images_current<${#STALE_IMAGES_CURRENT[@]} ; j_stale_images_current+=2 )) ; do
```
Comment (damccorm) [inline .test-infra/tools/stale_dataflow_prebuilt_image_cleaner.sh:108]: Why ware we double incrementing here?

**Review by damccorm — COMMENTED** [review]
One question, otherwise LGTM

File: .test-infra/tools/stale_dataflow_prebuilt_image_cleaner.sh (hunk 1 above)
Comment (Abacn) [reply .test-infra/tools/stale_dataflow_prebuilt_image_cleaner.sh:108]: Now STALE_IMAGES_CURRENT is of format (digest creation_year digest creation_year ...) so we do not need to make a second  docker API call (`docker inspect ...`) for the images of nonzero length (actual image)

Comment (damccorm) [reply .test-infra/tools/stale_dataflow_prebuilt_image_cleaner.sh:108]: Ah, I see - got it, thanks

**Review by damccorm — APPROVED** [review]

**Pull request merged**